frontend/components/puzzle1.jsx

```diff
@@ -9,8 +9,15 @@ class Puzzle1 extends React.Component {
     super(props);
     this.state = {
       value: "",
+      cipher: "",
     };
     this.handleChange = this.handleChange.bind(this);
+    this.cipherPing();
+
+    const { socket } = props;
+    socket.on('cipher_return', (data) => {
```

Comment:
lets set the listener before we call `cipherPing`, since, although unlikely, it's possible the response will come back before we set up the listener